src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1824,8 +1824,12 @@ static void updateAttrInNotifyCer
         if (caP->compoundValueP != NULL)
         {
           delete caP->compoundValueP;
+          caP->compoundValueP = NULL;
         }
-        caP->compoundValueP = targetAttr->compoundValueP == NULL ? NULL : targetAttr->compoundValueP->clone();
+
+        // Steal compound value from targetAttr
+        caP->compoundValueP        = targetAttr->compoundValueP;
```

Comment:
Actually, no strategic change is needed, but good that you wondered.
I found out that Metadata::release was empty.
Before implementing compound values for Metadata, there was nothing to delete. Now there is.